pennylane/qchem/tapering.py

```diff
@@ -538,3 +538,77 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
         tapered_hartree_fock.append(0)
 
     return np.array(tapered_hartree_fock).astype(int)
+
+
+def _is_commuting_obs(ham_a, ham_b, wire_map=None):
+    r"""Check for commutivity between two Pauli observables.
+
+    Args:
+        ham_a (Hamiltonian): first observable
+        ham_b (Hamiltonian): second observable
+
+    Returns:
+        Bool: representing whether ham_a and ham_b commutes or not.
+    """
+    for op1 in ham_a.ops:
+        for op2 in ham_b.ops:
+            if not qml.grouping.is_commuting(op1, op2, wire_map):
+                return False
+    return True
+
+
+def taper_excitations(generators, paulixops, paulix_sector, singles, doubles):
```

Comment:
This function is specifically tailored for tapering both single and double excitation operators simultaneously, which means users can't really leverage it to taper other gates.

Couldn't we use the same strategy as in here to taper an arbitrary gate? I think that would be very powerful

Reply:
Yep, this strategy can taper any arbitrary gate for which we have the generators.